File: Simplify Path/Simplify Path.py
class Solution:
    # @param path, a string
    # @return a string
    def simplifyPath(self, path):
        if path is None:
            return None
        
        List = path.split('/')
        stk = []
        
        for i in List:
            if i == '' or i == '.':
                continue
            elif i == '..':
                if stk:
                    stk.pop()
            else:
                stk.append(i)
                
        return '/' + '/'.join(stk)
                    

File: Simplify Path/Simplify Path.java
public class Solution {
    public String simplifyPath(String path) {
        String[] elements = path.split("/");
        List<String> res = new ArrayList<String>();
        
        for(String str : elements){
            if(str.equals(".") || str.equals(""))         continue;
            else if(str.equals("..")){
                if(!res.isEmpty()){
                    res.remove(res.size() - 1);
                }
            }
            else{
                res.add(str);
            }
        }
        
        StringBuilder ret = new StringBuilder();
        
        for(String str : res){
            ret.append("/");
            ret.append(str);
        }
        
        if(ret.toString().equals(""))       ret.append("/");
        
        return ret.toString();
    }
}
